haas/client.py

```diff
@@ -0,0 +1,801 @@
+from error_clientlib import *
+
+from haas.rest import APIError, ServerError
+    # Do we need ServerError ? It is not supposed to know 
+    # what happens on the server side
+import requests
+import os
+import urllib
+import json
+
+
+#Return Codes:
+""" Following is how return codes should be interpreted:
+    -1  :       Operation Failed
+    1   :       Operation was successful
+    2   :       Duplication error. Entity already exists
+    3   :
+
+"""
+
+
+
+class Haas:
+    """Client class for making Haas API calls.
+    Note that this client library is not yet complete and will be added to as
+    needed.
+
+    Example:
+        h = Haas(endpoint="http://127.0.0.1:5000")
+        h.node_connect_network("node-2000", "eth1", "production-network",
+        "vlan/native")
+
+    Errors are thrown when receiving HTTP status_codes from the HaaS server
+    that are not in the [200,300) range
+    """
+
+    def __init__(self, endpoint=None):
+        """Initiatlize an instance.
+
+        If endpoint is None, use the endpoint specification from:
+            1) The HAAS_ENDPOINT env variable or
+            2) Take the endpoint from [client] endpoint
+
+        Exceptions:
+            LookupError - no endpoint could be found
+        """
+        config.load(filename='haas.cfg')
+        if endpoint != None:
+            self.endpoint = endpoint
+        else:
+            self.endpoint = os.environ.get('HAAS_ENDPOINT')
+            if self.endpoint is None:
+                try:
+                    self.endpoint = cfg.get('client', 'endpoint')
+                except LookupError:
+                    print("no endpoint found")
+
+    def object_url(self, *args):
+        """Append the arguments to the endpoint URL"""
+        url = self.endpoint
+        for arg in args:
+            url += '/' + urllib.quote(arg,'')
+        return url
+
+    def user_create(self, username, password):
+        """Create a user <username> with password <password>."""
+        url = self.object_url('user', username)
+        data = {'password': password}
+        r = requests.put(url, data=json.dumps(data))
+
+        if (200 <= r.status_code < 300):
+            return                    #Operation successful
+        elif (r.status_code == 409):
+            raise DuplicateName("FAILURE: Name already exists" )    #Username already exists
+        elif(500 <= r.status_code):
+            raise ServersideError("FAILURE: Server failed to process the request ")
+        else:
+            raise UnknownError("FAILURE: Contact the HaaS Admin. Unknown error occured !!! ")      #Operation failed
+
+    def network_create(self, network, creator, access, net_id):
+        """Create a link-layer <network>. See docs/networks.md for details."""
+        url = self.object_url('network', network)
+        data = {'creator': creator, 'access': access, 'net_id': net_id}
+        r = requests.put(url, data=json.dumps(data))
+
+        if (200 <= r.status_code < 300):
```

Comment:
Same comment here as in the CLI: it might make sense to check the status code in the dispatcher function once for all of the client calls, then throw exceptions as necessary from there.